pkg/apis/vsphereprovider/v1alpha1/vsphereproviderconfig_types.go

```diff
@@ -52,8 +52,38 @@ type VSphereMachineProviderSpec struct {
 	// machine is cloned.
 	// +optional
 	DiskGiB int32 `json:"diskGiB,omitempty"`
+	// Snapshot is the name of the snapshot from which the VM was cloned
+	// +optional
+	Snapshot string `json:"snapshot"`
+
+	// CloneMode specifies the type of clone operation.
+	// The LinkedClone mode is only support for templates that have at least
+	// one snapshot. If the template has no snapshots, then CloneMode defaults
+	// to FullClone.
+	// When LinkedClone mode is enabled the DiskGiB field is ignored as it is
+	// not possible to expand disks of linked clones.
+	// Defaults to LinkedClone, but fails gracefully to FullClone if the source
```

Comment:
I can't really see where this "fails gracefully to FullClone", seems we are missing some logic in reconciler line 344